### PR TITLE
Testing component blocks

### PR DESCRIPTION
**Pull request by knownasilya**
Basically taken from https://gist.github.com/code0100fun/f5922b6f44eee2abc21d 

Thanks @code0100fun and https://emberweekend.firebaseapp.com/episodes/the-weekend-strikes-back

### CODE REVIEW

File: source/testing/testing-components.md [hunk 1]
```diff
@@ -201,6 +201,41 @@ test('trigger external action when button is clicked', function(assert) {
 });
 ```
 
+### Testing component blocks and defaults
+
+Often you might define a component which works in block form:
+
+```app/templates/components/my-component.hbs
+{{#if template}}
```
Comment (rwjblue) [inline source/testing/testing-components.md]: We intend to add `hasBlock?` helper for this, I am unsure that we should suggest this (though checking `template` will likely always work).

**Comment by trek** [comment]
Going to close from lack of reply/activity. Feel free to re-open if we can address @rwjblue's question.

**Comment by knownasilya** [comment]
What was the question? lol. Looks like we are waiting on `hasBlock?` to be implemented? Seems silly to wait on that and not have any documentation in the mean time. We can always update the docs once that has landed.

**Comment by rwjblue** [comment]
Accessing `this.get('template')` triggers a deprecation in 1.13, and `hasBlock` is enabled in 1.13.

**Comment by rwjblue** [comment]
Also, setting `template` does nothing in 1.13.

**Comment by knownasilya** [comment]
@rwjblue so just replace template with hasBlock?

**Comment by knownasilya** [comment]
What's the alternative to setting `template`?

**Comment by rwjblue** [comment]
> so just replace template with hasBlock?

Yes, without the question mark:

```hbs
{{#if hasBlock}}
  Stuff
{{/if}}
```

**Comment by knownasilya** [comment]
@rwjblue what about setting the template? What's the new API for that?

**Comment by rwjblue** [comment]
There isn't one.  The template for a component is supplied by the block provided by the calling location.

Please explain the intent of changing the template of a component, maybe I have a better answer/explaination...

**Comment by knownasilya** [comment]
For testing integration between different components, e.g.

```hbs
{{#video-player as |playing|}}
  {{ad-overlay ad=myAd playing=playing}}
{{/video-player}}
```

Or even static data, since you want to test if your component is behaving correctly when in block mode.

**Comment by rwjblue** [comment]
For testing, you should be using the newer style component integration tests.  We need more documentation for it though.

Added in https://github.com/switchfly/ember-test-helpers/pull/38, you can do something like this [example from liquid-fire](https://github.com/ef4/liquid-fire/blob/new-test-helpers/tests/integration/helpers/liquid-if-test.js).

**Pull request closed without merging**